File: LeetCode/Prep/Top_interview_questions_easy/Arrays/bestTimeToBuyAndSellStockTwo.java
class Solution {
    public int maxProfit(int[] prices) {
        int i=0;
        int b=prices[0];
        int s=prices[0];
        int total = 0;
        while(i<prices.length-1){
            while(i<prices.length-1 && prices[i]>=prices[i+1])
                i++;
            b=prices[i];
            
            while(i<prices.length-1 && prices[i]<=prices[i+1])
                i++;
            s=prices[i];
            total+=s-b;
            
        }
        
        return total;
        
    }
}

File: LeetCode/Prep/Top_interview_questions_easy/Arrays/bestTimeToBuyAndSellStockTwo.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        i=0
        b=prices[0]
        s=prices[0]
        total =0
        while i<len(prices)-1:
            while i<len(prices)-1 and prices[i]>=prices[i+1]:
                i+=1
            b=prices[i]
            while i<len(prices)-1 and prices[i]<=prices[i+1]:
                i+=1
            s=prices[i]
            total+= s-b
        
        return total
